Automattic/jetpack
# Jetpack Connection: Add stricter check before updating 'jetpack_connection_active_plugins' option

When checking if the `jetpack_connection_active_plugins` option needs to be updated, we used to rely on the difference between the counts of the stored option and the value of `Plugin_Storage::$plugins`, which is set on `plugins_loaded`.

This PR adds a stricter check, which relies on the exact match between plugin slugs and covers the following scenario:
It's possible that first a Jetpack connected plugin is deactivated via `cli` and then another standalone Jetpack plugin is activated via `cli`.

Whenever the next `POST` request occurs, the count of configured plugins will be the same with the count of stored plugins (while the actual values will differ) therefore the option will not be updated.

## Proposed changes:

* `Automattic\Jetpack\Connection\Plugin_Storage`: Update `maybe_update_active_connected_plugins` method with a stricter check between the `jetpack_connection_active_plugins` option and the value of `Plugin_Storage::$plugins` to determine if the option needs to be updated

### Other information:


## Jetpack product discussion


pet6gk-19m-p2#comment-1040

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Ensure you have at least two connected Jetpack plugins present at your site. One should be active while the other not. Example: Jetpack plugin active, Boost plugin inactive
* Via `cli` activate the inactive one and deactivate the active one. Example:
```
wp plugin activate jetpack-boost
wp plugin deactivate jetpack
```
* Visit any admin page to trigger a `POST` request
* Check the value of `jetpack_connection_active_plugins` option: Without the patch, the value will not have been updated. With the patch, the value will be updated.

* Check the normal flows just in case too. This means activating/deactivating Jetpack plugins manually and ensuring `jetpack_connection_active_plugins` reflects the change
